src/playground-code-editor.ts

```diff
@@ -447,7 +451,14 @@ export class PlaygroundCodeEditor extends LitElement {
             new CustomEvent('request-completions', {
                 detail: {
                     isRefinement,
-                } as CodeEditorChangeData,
+                    fileName: this.fileName,
+                    fileContent: this.value,
+                    tokenUnderCursor,
+                    cursorIndex: this.cursorIndex,
+                    provideCompletions: (completions: EditorCompletion[]) => {
+                        this.completions = completions;
```

Comment:
I think a race could happen here, where e.g. completion request 1 and 2 are sent, but the answers are received in the reverse order. Once we've sent request 2, we should ignore any response for request 1.

One way to do this would be something like:

```ts
private _currentCompletionRequestId = 0;
...
const id = ++this._currentCompletionRequestId;
const provideCompletions = (completions: EditorCompletion[]) => {
  if (id === this._currentCompletionRequestId) {
    this.completions = completions;
  }
};
```